magento/magento2
# Cleaning caches send all tags at once, not one by one

### Description (*)

MView updates performed by the `indexer_update_all_views` cron some times take very long time with seemingly no load on the PHP or DB. Detailed investigation reveled that the reason for long execution times especially on larger data sets (100K+ products) are due to the fact that some times cache flush requests are sent in separate request for each product tag.

This fix combines all relevant tags in one request and makes MView update execute significantly faster.  

### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Ensure that indexers are set to update on schedule
2. Ensure that all indexes are in fine order for cleaner test (run `indexer:status`, to check) 
3. Alter data for several products, e.g. by running. This assumes price attribute id is 77 
```UPDATE catalog_product_entity_decimal SET value=2255.00 WHERE attribute_id=77 AND entity_id<5000;```
4. Ensure that crons are running and are executed. Can check with `ps aux | grep index`
5. Wait for MViews to be updated so that `indexer:status` shows that all indexes are are fine
6. Check site and see that caches were flushed.

### Questions or comments


### Contribution checklist (*)
 - [+] Pull request has a meaningful description of its purpose
 - [+] All commits are accompanied by meaningful commit messages
 - [+] All new or changed code is covered with unit/integration tests (if applicable)
 - [+] All automated tests passed successfully (all builds are green)
